src/infoview.ts

```diff
@@ -108,6 +110,10 @@ export class InfoProvider implements Disposable {
         }
     }
 
+    onNewTacticState(handler_name: string, handler: (state: string, widget: object) => void): void {
```

Comment:
```suggestion
    onNewTacticState(handler_name: string, handler: (loc: Location, state: InfoResponse) => void): void {
```

The `widget` field doesn't really contain any information (it's an id that you pass to the get_widget command).  You probably also want the location (filename & position) so that you can insert something there.

Can you explain what the `handler_name` should be?  Is this the name of the other extension?

I'm not sure if it's possible, but the nicest way would be to use the `EventEmitter` class from vscode:
```typescript
    private onNewTacticStateEmitter = new EventEmitter<{ location: Location, info: InfoResponse }>();
    onNewTacticState = this.onNewTacticStateEmitter.event;
```